projects/igniteui-angular/src/lib/grids/cell.component.ts

```diff
@@ -244,7 +239,7 @@ export class IgxGridCellComponent implements OnInit, OnChanges, OnDestroy, CellT
      * @memberof IgxGridCellComponent
      */
     public get template(): TemplateRef<any> {
-        if (this.editMode) {
+        if (this.editMode && this.formGroup) {
```

Comment:
Yes. It was a strange timing/change detection mismatch in one of the test scenario that I could not reproduce outside of the tests:
https://dev.azure.com/IgniteUI/igniteui-angular/_build/results?buildId=42173&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=59a85588-b0ba-5043-24c4-d9e29d89c6f6&l=7086

For some reason when forcing editing with:
cell.editMode = true;
And then doing changing the data view so that this cell now has a new data context and should not be in editMode, in the test it still tried to render it as if it is in edit mode, hence the error.